design/experimental-new-features-in-modular.md

```diff
@@ -0,0 +1,204 @@
+# Shipping experimental new features in Modular
+
+## Background
+
+As the Azure SDK for JavaScript continues to evolve, it is important for us to have a dedicated space to ship experimental features within our own client or language. By providing a separate subpath export for experimental features alongside with the regular release of Modular, we can gather valuable feedback from our customers who have tried these new features. This approach enables us to receive early insights and iterate on the experimental features while ensuring that our mainline releases remain stable and reliable.
+
+## Questions
+
+1. **Version picking strategy**  
+  One latest GA + One latest preview version.  
+  If the latest preview version < latest GA version, then we will use latest GA version as the experiment version.  
+  _Questions_:
+    1. do we allow customers to set their own preview version?
+
+1. **Feature picking strategy**  
+  Let's say we have Feature 1, 2, 3, 4 to experiment
+  _Questions_:
+    1. Is it possible that we want features 1, 2 in one RP but features 3, 4 in another RP?
+    1. If service team want a pure preview with their service version without any client side preview features, is this allowed?  
+
+1. **Feature staging status change**  
+  _Questions_:
+   1. Is it possible that a feature could be GA or withdrew based on customers feedback?
+   1. how are we going to decide, statistically?
+   1. If a feature goes GA, we shall refresh all the packages that has the experimented feature, or just refresh all the packages based on release requirement?
+   1. If a feature goes withdraw, we shall refresh all the packages that has the experiment feature?
+
+1. **Feature collisions**
+   1. Is it possible that one feature may have part of the code collisions with the other feature?
+   1. Is it possible that experimental features may have part of the code collisions with the stable code?
```

Comment:
I think from the SDK perspective the most important distinction is just which features are experimental/beta vs. stable, and I think we have to assume they won't conflict with each other in such a way that it'll make it impossible to generate a client that can use all the experimental features at the same time.